### PR TITLE
CBG-2761: return bad request when writing read_only collection_access properties

### PR DESCRIPTION
**Pull request by mohammed-madi**
CBG-2761

Describe your PR here...
- Return 400 bad request when writing read_only collection_access properties for /_user and /_role endpoints

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1609/

### CODE REVIEW

File: rest/user_api_test.go [hunk 1]
```diff
@@ -1369,9 +1369,10 @@ func TestGetUserCollectionAccess(t *testing.T) {
 	collectionPayload := fmt.Sprintf(`,"%s": {
 					"admin_channels":["foo", "bar1"]
 				}`, collection2Name)
-	//rdOnlycollectionPayload := fmt.Sprintf(`,"%s": {
-	//				"jwt_channels":["foo", "bar1"]
-	//			}`, collection2Name)
+	rdOnlycollectionPayload := fmt.Sprintf(`,"%s": {
+					"jwt_channels":["foo", "bar1"]
+				}`, collection2Name)
```
Comment (torcolvin) [inline rest/user_api_test.go]: can you write a loop for each unmodifable property?

File: rest/utilities_testing.go [hunk 2]
```diff
@@ -1007,6 +1006,17 @@ func (rt *RestTester) SetAdminChannels(username string, keyspace string, channel
 	}
 
 	currentConfig.SetExplicitChannels(*scopeName, *collectionName, channels...)
+	// Remove read only properties returned from the user api
+	for _, scope := range currentConfig.CollectionAccess {
+		if scope != nil {
+			for _, collectionAccess := range scope {
+				collectionAccess.Channels_ = nil
```
Comment (torcolvin) [inline rest/utilities_testing.go:1013]: which tests are setting this property? is this just a result of marshal / unmarshalling?

Comment (mohammed-madi) [reply rest/utilities_testing.go:1013]: TestMultiCollectionChangesUserDynamicGrant and TestMultiCollectionChangesUserDynamicGrantDCP
It's included in the user api response, which this function unmarshals, modifies then marshals and uses it in line 1022

Comment (torcolvin) [reply rest/utilities_testing.go:1013]: I think I'm OK with this - I don't think adding the boilerplate to those tests would be helpful.

**Review by torcolvin — APPROVED** [review]

**Pull request merged**